forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/repository/migration/v3_1_6/MongoDbPaymentsChangeLog.java

```diff
@@ -24,83 +24,153 @@
+        elapsedTime.stop();
+        log.info(">>>>>>>>>> Upgraded [{}] {} objects in {} seconds.", entriesModified, clazz.getName(), elapsedTime.getTotalTimeSeconds());
+        log.info("-----------------------------------------------------------------------");
+    }
 
-        CloseableIterator<FRInternationalScheduledPaymentSubmission2> frInternationalScheduledPaymentSubmissions = getLegacyDocuments(mongoTemplate, FRInternationalScheduledPaymentSubmission2.class);
-        frInternationalScheduledPaymentSubmissions.forEachRemaining(d -> migrate(mongoTemplate, d, toFRInternationalScheduledPaymentSubmission(d)));
+    /*
+     * List of objects to update
+     * @return list of objects to update
+     */
+    private List<ObjectToUpdate> getObjectsToUpdate(){
+        List<ObjectToUpdate> objectsToUpdate = new ArrayList<>();
+        // Domestic payments
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticConsent.class, "domesticConsent"));
+
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticPaymentSubmission.class, "domesticPayment"));
+        // Domestic scheduled payments
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticScheduledConsent.class, "domesticScheduledConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticScheduledPaymentSubmission.class, "domesticScheduledPayment"));
+        // Domestic standing orders
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticStandingOrderConsent.class, "domesticStandingOrderConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRDomesticStandingOrderPaymentSubmission.class, "domesticStandingOrder"));
+        // International payments
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalConsent.class,"internationalConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalPaymentSubmission.class, "internationalPayment"));
+        // International scheduled payments
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalScheduledConsent.class, "internationalScheduledConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalScheduledPaymentSubmission.class, "internationalScheduledPayment"));
+        // International standing Orders
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalStandingOrderConsent.class, "internationalStandingOrderConsent"));
+        objectsToUpdate.add(new ObjectToUpdate(FRInternationalStandingOrderPaymentSubmission.class, "internationalStandingOrder"));
+
+        objectsToUpdate.add(new ObjectToUpdate(FRPaymentSetup.class, "paymentSetupRequest"));
+
+
+        return objectsToUpdate;
+    }
 
-        CloseableIterator<FRInternationalStandingOrderConsent3> frInternationalStandingOrderConsents = getLegacyDocuments(mongoTemplate, FRInternationalStandingOrderConsent3.class);
-        frInternationalStandingOrderConsents.forEachRemaining(d -> migrate(mongoTemplate, d, toFRInternationalStandingOrderConsent(d)));
+    class ObjectToUpdate{
```

Comment:
Minor - you could make this static and declare legacy class as `Class<? extends LegacyCountrySubDivision>`.